samples/TerraFX/Graphics/HelloSmoke.cs

```diff
@@ -0,0 +1,364 @@
+            {
+                var indexBuffer = graphicsContext.Device.MemoryAllocator.CreateBuffer(GraphicsBufferKind.Index, GraphicsResourceCpuAccess.None, sizeof(ushort) * 6);
+                var pIndexBuffer = indexStagingBuffer.Map<ushort>();
+
+                pIndexBuffer[0] = 0; // a
+                pIndexBuffer[1] = 1; // b
+                pIndexBuffer[2] = 2; // d
+
+                pIndexBuffer[3] = 0; // b
+                pIndexBuffer[4] = 2; // c
+                pIndexBuffer[5] = 3; // d
+
+                indexStagingBuffer.Unmap(0..(sizeof(ushort) * 6));
+                graphicsContext.Copy(indexBuffer, indexStagingBuffer);
+
+                return indexBuffer;
+            }
+
+            static GraphicsBuffer CreateConstantBuffer(GraphicsContext graphicsContext)
+            {
+                var constantBuffer = graphicsContext.Device.MemoryAllocator.CreateBuffer(GraphicsBufferKind.Constant, GraphicsResourceCpuAccess.Write, 256);
+
+                var pConstantBuffer = constantBuffer.Map<Matrix4x4>();
+                pConstantBuffer[0] = Matrix4x4.Identity;
+                constantBuffer.Unmap(0..sizeof(Matrix4x4));
+
+                return constantBuffer;
+            }
+
+            static GraphicsTexture CreateTexture3D(GraphicsContext graphicsContext, GraphicsBuffer textureStagingBuffer, bool isQuickAndDirty)
+            {
+                const uint TextureWidth = 256;
+                const uint TextureHeight = 256;
+                const ushort TextureDepth = 256;
+                const uint TextureDz = TextureWidth * TextureHeight;
+                const uint TexturePixels = TextureDz * TextureDepth;
+                const uint TextureSize = TexturePixels * 4;
+
+                var texture3D = graphicsContext.Device.MemoryAllocator.CreateTexture(GraphicsTextureKind.ThreeDimensional, GraphicsResourceCpuAccess.None
+                    , TextureWidth, TextureHeight, TextureDepth
+                    , texelFormat: TexelFormat.RGBA4xByte);
+                var pTextureData = textureStagingBuffer.Map<UInt32>();
+
+                var random = new Random(Seed: 1);
+
+                // start with random speckles
+                for (uint n = 0; n < TexturePixels; n++)
+                {
+                    // convert n to indices
+                    float x = n % TextureWidth;
+                    float y = (n % TextureDz) / TextureWidth;
+                    float z = n / TextureDz;
+                    // convert indices to fractions in the range [0, 1)
+                    x /= TextureWidth;
+                    y /= TextureHeight;
+                    z /= TextureHeight;
+                    // make x,z relative to texture center
+                    x -= 0.5f;
+                    z -= 0.5f;
+                    // get radius from center, clamped to 0.5
+                    float radius = MathF.Abs(x); // MathF.Sqrt(x * x + z * z);
+                    if (radius > 0.5f)
```

Comment:
The repo always uses braces to help avoid bugs.
```suggestion
                    if (radius > 0.5f)
                    {
```

Reply:
ok